src/test/java/com/airlines/dao/impl/CrewDaoImplTest.java

```diff
@@ -0,0 +1,182 @@
+
+    private static DataSource dataSource;
+
+    @BeforeClass
+    public static void init() {
+        dataSource = TestDatasourceProvider.createTestDatasource();
+        crewDao = new CrewDaoImpl(dataSource);
+    }
+
+    @Before
+    public void dataBaseInit() throws SQLException {
+        createTables();
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewMemberProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(null, null));
+        assertEquals("Valid crewMember entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfInvalidArgumentCrewProvidedToAddMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.add(CREW_MEMBER, null));
+        assertEquals("Valid crew entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldCorrectlyAddCrewMemberToCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertTrue(list.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldGetCrewMembersByCrewId() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertEquals(1, list.size());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewIsNotProvidedForGetByIdMethod() throws SQLException {
+        Crew invalidCrew = buildInvalidCrewFromDataSet();
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.getByCrewId(invalidCrew.getId()));
+        assertEquals("Cannot find crew members for crew without id", exception.getMessage());
+    }
+
+    @Test
+    public void getByCrewName() throws SQLException {
+        populateTestData();
+        List<CrewMember> list = crewDao.getByCrewName("Vityaz");
+        assertEquals(3, list.size());
+    }
+
+    @Test
+    public void shouldRemoveCrewMemberFromCrew() throws SQLException {
+        populateTestData();
+        crewDao.add(CREW_MEMBER, CREW);
+        List<CrewMember> list = crewDao.getByCrewId(3L);
+        assertTrue(list.contains(CREW_MEMBER));
+        crewDao.removeCrewMemberFromCrew(CREW, CREW_MEMBER);
+        List<CrewMember> listAfterRemove = crewDao.getByCrewId(3L);
+        assertFalse(listAfterRemove.contains(CREW_MEMBER));
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewIsNotProvidedForRemoveMethod() {
+        IllegalArgumentException exception = assertThrows(IllegalArgumentException.class, () -> crewDao.removeCrewMemberFromCrew(INVALID_CREW, CREW_MEMBER));
+        assertEquals("Valid crew entity should be provided", exception.getMessage());
+    }
+
+    @Test
+    public void shouldThrowExceptionIfCrewMemberIsNotProvidedForRemoveMethod() {
```

Comment:
do the same as for ADD method, cover 2 conditions more, when CREW is null and CREW_MEMBER has ID = null